### PR TITLE
Fix behavior when running as normal user 

### PR DESCRIPTION
**Pull request by jyeray**

### CODE REVIEW

File: test/client/client_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2016 SUSE LLC.
```
Comment (ancorgs) [inline test/client/client_spec.rb]: The file should be called `test/client/alternatives_spec.rb`. Isn't it?

File: src/clients/alternatives.rb [hunk 2]
```diff
@@ -1,3 +1,12 @@
 require "y2_alternatives/dialog/list_alternatives"
 
-Y2Alternatives::Dialog::ListAlternatives.run
+Yast.import "Confirm"
+
+# Checks if user is root and create a ListAlternatives dialog
+class AlternativesClient
```
Comment (ancorgs) [inline src/clients/alternatives.rb]: I would prefer to have this in a class `Y2Alternatives::Client::Alternatives` (or any other name), with this file containing only the call.

File: test/client/client_spec.rb (hunk 1 above)
Comment (ancorgs) [inline test/client/client_spec.rb]: Well, it should if you change the implementation as suggested above (with a separate class in `/lib`).

File: src/lib/y2_alternatives/dialog/list_alternatives.rb [hunk 3]
```diff
@@ -104,10 +104,14 @@ def update_table(index)
       end
 
       def accept_handler
-        @alternatives_list.each(&:save)
+        @alternatives_list.each(&:save) if permissions?
         finish_dialog
       end
 
+      def permissions?
```
Comment (ancorgs) [inline src/lib/y2_alternatives/dialog/list_alternatives.rb]: I also wonder if this logic belongs to the dialog. We have other options:
- The rails way: `Alternative#save` runs all checks and handles the possible exceptions coming from `Y2Alternatives::Control` returning false if something was wrong (or true otherwise).
- Using exceptions: `Alternative#save` checks permissions and raises an exception if needed.
- ...

In both cases, checking `Process::UID` will be done in the `Alternative` class and not in the dialog. It does not belong here.

Comment (jreidinger) [inline src/lib/y2_alternatives/dialog/list_alternatives.rb]: rails way is actually both, `save` and `save!` with first one returning false and second one raising exception.

File: src/lib/y2_alternatives/alternative.rb [hunk 4]
```diff
@@ -129,7 +129,12 @@ def automatic_mode!
 
     def save
       return unless @modified
-      STATUS_COMMANDS[@status].execute(self)
+      begin
+        STATUS_COMMANDS[@status].execute(self)
+        true
+      rescue Cheetah::ExecutionFailed
```
Comment (jreidinger) [inline src/lib/y2_alternatives/alternative.rb]: I think this cannot happen as Execute module itself already catch exceptions. of course unless you use cheetah directly.

File: src/clients/alternatives.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
-require "y2_alternatives/dialog/list_alternatives"
+require "y2_alternatives/client/start_checking_permissions"
 
-Y2Alternatives::Dialog::ListAlternatives.run
+Y2Alternatives::Client::StartCheckingPermissions.new.main
```
Comment (jreidinger) [inline src/clients/alternatives.rb:3]: feels a bit strange about client name, but currently do not have in mind better name .. maybe `Client::Entry` but not sure. so just note, not blocking merge.

File: src/lib/y2_alternatives/client/start_checking_permissions.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+require "y2_alternatives/dialog/list_alternatives"
+
+Yast.import "Confirm"
+
+module Y2Alternatives
+  module Client
+    # Checks if user is root and create a ListAlternatives dialog
+    class StartCheckingPermissions
+      def main
+        Y2Alternatives::Dialog::ListAlternatives.run if Yast::Confirm.MustBeRoot
```
Comment (jreidinger) [inline src/lib/y2_alternatives/client/start_checking_permissions.rb]: in fact first `Y2Alternatives::` is not needed as code is already in this namespace, but it is just style taste. No strong opinion. so just note.

File: test/alternative_spec.rb [hunk 7]
```diff
@@ -176,33 +176,65 @@
...
+    context "if the alternative is modificated" do
       subject(:alternative) { editor_alternative_automatic_mode }
-      it "execute a SetChoiceCommand" do
+      before do
         alternative.choose!("/usr/bin/nano")
-        expect(Y2Alternatives::Control::SetChoiceCommand).to receive(:execute).with(alternative)
+      end
+
+      it "checks if user is root" do
+        expect(Process::UID).to receive(:rid)
```
Comment (jreidinger) [inline test/alternative_spec.rb]: missing return value, so test simulate that rid return nil, which is wrong.

File: test/client/alternative_spec.rb [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require_relative "../spec_helper.rb"
+require "y2_alternatives/dialog/list_alternatives"
+require "y2_alternatives/client/start_checking_permissions"
+
+describe Y2Alternatives::Client::StartCheckingPermissions do
```
Comment (jreidinger) [inline test/client/alternative_spec.rb]: this is wrong. file indicate that it testing code in client/alternative.rb, but here you describe code in `lib/y2_alternatives/client/start_checking_permissions.rb`

File: src/clients/alternatives.rb (hunk 5 above)
Comment (jyeray) [inline src/clients/alternatives.rb:3]: Well, I know that isn't a good name for a class, but I don't find any better.

**Comment by jreidinger** [comment]
LGTM
